src/middlewared/middlewared/plugins/pool_/dataset.py

```diff
@@ -579,6 +579,14 @@ async def do_create(self, app, data):
         if not inherit_encryption_properties:
             encryption_dict = {'encryption': 'off'}
 
+        if parent_ds['encrypted'] and data['encryption'] is False and not inherit_encryption_properties:
```

Comment:
We probably have to check the entire ancestor tree. (i.e. encrypted->parentA->childA)